components/global/mixer-adapter/cmd/mixer-adapter/main.go

```diff
@@ -65,40 +83,126 @@ func main() {
 		}
 	}
 
-	/* Mutual TLS feature to secure connection between workloads
-	   This is optional. */
-	adapterCertificate := os.Getenv(grpcAdapterCertificatePath) // adapter.crt //change the name
-	adapterprivateKey := os.Getenv(grpcAdapterPrivateKeyPath)   // adapter.key
+	// Mutual TLS feature to secure connection between workloads. This is optional.
+	adapterCertificate := os.Getenv(grpcAdapterCertificatePath) // adapter.crt
+	adapterPrivateKey := os.Getenv(grpcAdapterPrivateKeyPath)   // adapter.key
 	caCertificate := os.Getenv(caCertificatePath)               // ca.pem
+
 	spServerUrl := os.Getenv(spServerUrlPath)
 
+	directory := os.Getenv(directory)
+	waitingSec, _ := strconv.Atoi(os.Getenv(waitingTimeSecEnv))
+	waitingSize, _ := strconv.Atoi(os.Getenv(waitingSizeEnv))
+	tickerSec, _ := strconv.Atoi(os.Getenv(tickerSecEnv))
+
+	vendor := os.Getenv(vendorEnv)
+	dsn := os.Getenv(dsnEnv)
+
+	persist, err = strconv.ParseBool(os.Getenv(shouldPersist)) // this should be a string contains a boolean, "true" or "false"
+	if err != nil {
+		logger.Warnf("Could not convert env of persisting : %s", err.Error())
+		persist = false
+	}
+
 	logger.Infof("Sp server url : %s", spServerUrl)
+	logger.Infof("Should persist : %s", persist)
```

Comment:
Either validations or suitable default values should be used for mandatory parameters looked up as env variables.